Backend/BlockChangesService.cs

```diff
@@ -45,12 +44,18 @@ public Task<List<BlockChangeRequest>> FindAllStudentRequests(Guid studentId)
             return _blockChangesCollection.Find(x => x.StudentId == studentId).ToListAsync();
         }
 
-        public async Task MakeExchangeAndDeleteRequests(BlockChangeRequest from, BlockChangeRequest to)
+        public async Task<bool> MakeExchangeAndDeleteRequests(BlockChangeRequest request)
         {
-            await SetDoneStatus(from);
-            await SetDoneStatus(to);
-            await RemoveStudentRequests(from.StudentId, from.BlockFrom.CourseId);
-            await RemoveStudentRequests(to.StudentId, to.BlockFrom.CourseId);
+            var requestForExchange = await FindExchange(request);
+            if (request == null)
+            {
+                return false;
+            }
+            await SetDoneStatus(request);
+            await SetDoneStatus(requestForExchange);
+            await RemoveStudentRequests(request.StudentId, request.BlockFrom.CourseId);
```

Comment:
You will remove all requests for the course? May be you should remove just request for the course and the block (what if there are more blocks for the same course)

Reply:
OK